### ISSUE TITLE
Icon Request: Debug Actions (fa-step-into, fa-step-out, fa-step-over, fa-next-line)

### ISSUE THREAD

**Issue by SylvainCorlay** [FortAwesome/Font-Awesome#6619]
Icon name  | Example
------------- | -------------
fa-step-into  | ![step-into](https://cloud.githubusercontent.com/assets/2397974/7914624/379f77a4-0846-11e5-9c38-2ca99a322d25.png)
fa-step-out |  ![step-out](https://cloud.githubusercontent.com/assets/2397974/7914628/3ba4b0c6-0846-11e5-9966-0a60e95ec898.png)
fa-step-over | ![step-over](https://cloud.githubusercontent.com/assets/2397974/7914633/3d7a778c-0846-11e5-81a5-b6f1a3fa2f8f.png)
fa-next-line | ![next-line](https://cloud.githubusercontent.com/assets/2397974/7914684/820724c2-0846-11e5-8057-c0702b9ccd7e.png)

I have svg and glyph versions of those if you are willing to use this design. (should be pixel perfect at 14px).

**Comment by sccolbert** [comment]
+1 nice!

**Comment by jdfreder** [comment]
Well drawn @SylvainCorlay , these look slick!  :+1:

**Comment by Carreau** [comment]
+1. How does step-in/out look like with only 3 lines ?

**Comment by SylvainCorlay** [comment]
@Carreau Here are the 3-lines versions of step-into and step-out:

Icon name  | Proposal
------------- | -------------
fa-step-into  | ![step-into-3](https://cloud.githubusercontent.com/assets/2397974/7923223/6bbb359e-087f-11e5-8a91-e98d899572f7.png)
fa-step-out | ![step-out-3](https://cloud.githubusercontent.com/assets/2397974/7923266/9f3b2db6-087f-11e5-8032-145437cc49f3.png)

**Comment by sccolbert** [comment]
but those are only 4 line high instead of 5 (like step over). I prefer them all equal overall height.

**Comment by Carreau** [comment]
> but those are only 4 line high instead of 5 (like step over). I prefer them all equal overall height.

You also need to think if they are next to each other  in a button group horizontally. Having the 3 lines aligned across button make more sens (IMHO).

**Comment by sccolbert** [comment]
@Carreau good point. Would be useful to see them all arranged horizontally and pick the combo which is most visually pleasing.

**Comment by SylvainCorlay** [comment]
Here is the Spyder debug toolbar with the font-awesome theme and those icons.
![spyder_toolbar](https://cloud.githubusercontent.com/assets/2397974/7924832/9e9326ae-088c-11e5-812d-82e668b0f9d0.png)

**Comment by sccolbert** [comment]
After seeing that, I think I favor the idea of @Carreau to have 3-line step-in and step-out buttons so that the content lines will align horizontally.

**Comment by cjrh** [comment]
👍

**Comment by goanpeca** [comment]
:+1: I also favor the idea of three lines aligned horizontally, would probably be more pleasing... i.e.

![test](https://cloud.githubusercontent.com/assets/3627835/7931219/039db2f8-090f-11e5-9cce-1f1fdf9b64ff.png)

**Comment by ccordoba12** [comment]
+1 to this and also to @Carreau idea about using only 3 lines.

**Comment by Nodd** [comment]
Is it possible to have the arrow head of fa-step-out a bit further on the left ? This way it would be more obvious that the debug is going out of the bloc.
Same goes for fa-step-into, the start of the arrow should be shorter.
In both cases the point is that the arrow itself should show that the start and end point are at different alignment.

**Comment by kwendenarmo** [comment]
+1

**Comment by SylvainCorlay** [comment]
The latest screenshot of Spyder implementing the suggestions above:
![main-toolbar](https://cloud.githubusercontent.com/assets/2397974/7987393/2ec1440c-0aab-11e5-9109-7656c5c32a36.png)

**Comment by sccolbert** [comment]
@SylvainCorlay lgtm!

**Comment by Nodd** [comment]
Not my suggestion... :( Here is what I meant:
![fa-setp-into](https://cloud.githubusercontent.com/assets/506348/8003046/5fefe6cc-0b74-11e5-923e-47203204300c.png)
![fa-setp-out](https://cloud.githubusercontent.com/assets/506348/8003053/655b9e94-0b74-11e5-8db6-08b883e7f208.png)

**Comment by blink1073** [comment]
+1 to latest icons

**Comment by blacatena** [comment]
+1

**Comment by ssunkara1** [comment]
+1

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂
